### DISCUSSION TITLE
Attempting to disable a task results in COMException.

### DISCUSSION THREAD

**Discussion by Monsoonexe** [dahall/TaskScheduler#945] [Q&A] [answered]
I used a program written with this library to create a task (and it works). However, now I want to use a different program written with this library to disable (or delete) that same task. However, I'm getting a "System.Runtime.InteropServices.COMException (0x80070534): (11, 38).

Between then and now, I have renamed my machine, but there has only ever been one user and the username hasn't changed.

Disable code:
`      using (var taskService = new TaskService())
      {
        var task = taskService.GetTask(taskName);
        task.Definition.Settings.Enabled = false;
        task.RegisterChanges(); //COMException thrown here
      }
`

If I try to delete the task, then I get an UnauthorizedAccessException.
Delete code:
`
  private static void DeleteTask()
  {
    Console.WriteLine($"Deleting {taskName}");
    using (var taskService = new TaskService())
    {
      //var task = taskService.GetTask(taskName);
      try
      {
        taskService.RootFolder.DeleteTask(taskName);
        Console.WriteLine("Done!");
      }
      catch (COMException e)
      {
        Console.WriteLine($"Failed to delete task: {e.Message} | {e.ErrorCode}");
      }
    }
  }`

Maybe I'm using the library wrong. Any help would be appreciated. Thanks for listening.

**Comment by dahall** [comment, marked as answer]
That error is ERROR_NONE_MAPPED: No mapping between account names and security IDs was done. See [this TechNet article](https://social.technet.microsoft.com/Forums/windows/en-US/d9f9a248-61b0-45fc-83be-22e5044300ba/no-mapping-between-account-names-and-security-ids-was-done-exception-from-hresult-0x80070534?forum=winserverDS) for some insight.

Are you able to view and edit the task in the Task Manager UI? (system app)

To disable a task you can simply write `TaskService.Instance.GetTask(taskName).Enabled = false;` Maybe this won't throw an exception.

Your exception on deletion is likely due to not running the code in Administrator mode.

**Reply by Monsoonexe** [reply]
Thanks for the response. Unfortunately that code snippet didn't work. I got an "UnauthorizedAccessException: Access is denied (0x80070005)". This is strange since my machine only has one user, and it has admin privileges. I am able to view and edit the task in the Task Manager, I just need some code to automate this process on many machines.

I think the problem is that I made this task under a different user, whose account was later deleted. I know what the domain and password used to be. I also notice that the "Author" field of the task I want to delete is empty (not sure if this is signifigant).

**Reply by dahall** [reply]
Even if you are Admin on the box, you have to specifically run your code as Admin (to bypass the UAC) starting the exe from an elevated commandline or using "Run as Administrator" when launching from Explorer.
